### PR TITLE
Beging to work on reseting password using AWS Cognito

### PR DESCRIPTION
**Pull request by jonwuzulike**
Users can reset their password using the "Forgot Password" link on the login page. The links navigates to the Forgotten Password Component from which users enter their password. From here the user enters their password and then clicks the "Reset Password" button. Clicking this button invokes the rest Password method with calls the Cognito Service 
image
image
image
image

### CODE REVIEW

**Comment by MarcinSalamon** [comment]
What's the progress?

**Comment by danielsidhu** [comment]
I'm not really seeing the changes besides 4 lines of code here.

**Comment by jonwuzulike** [comment]
You should be able to see the blocks of code in the images now.

**Comment by MarcinSalamon** [comment]
I can see the blocks of code, but I want to see code pushed to this branch.

**Comment by jonwuzulike** [comment]
Yes I just did a push this morning.

File: src/app/routes/routes.ts [hunk 1]
```diff
@@ -53,5 +54,9 @@ export const appRoutes: Routes = [
     {
         path: 'curriculum/dialog-view',
         component: DialogViewComponent
+    },
+    {
+        path: 'forgotten/password',
+        component: ForgottenPasswordComponent
```
Comment (MarcinSalamon) [inline src/app/routes/routes.ts:60]: routes.ts doesn't exist anymore, put this code in routes.module

File: src/app/services/cognito.service.ts [hunk 2]
```diff
@@ -116,7 +118,39 @@ export class CognitoService {
...
+     inputVerificationCode() {
+         var verificationCode = prompt('Please input verification code ' ,'');
+         var newPassword = prompt('Enter new password ' ,'');
+         cognitoUser.confirmPassword(verificationCode, newPassword, this);
+     }
+ });
+
+ }
+
+
```
Comment (MarcinSalamon) [inline src/app/services/cognito.service.ts]: too many white spaces

File: src/app/routes/forgotten-password/forgotten-password.component.html [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+      <mat-card-content>
+        <label>Enter your email address:</label>
+        <br>
+        <input type="email" [(ngModel)]="email">
+        <br>
+        <button mat-raised-button (click)="reset()">Reset Password</button>
+      </mat-card-content>
+    </mat-card>
+  </div>
+</div>
```
Comment (MarcinSalamon) [inline src/app/routes/forgotten-password/forgotten-password.component.html]: put in the end line

**Review by MarcinSalamon — CHANGES_REQUESTED** [review]
almost there, change the routes file to routes.module

File: src/app/routes/routes.ts (hunk 1 above)
Comment (jonwuzulike) [reply src/app/routes/routes.ts:60]: Completed.

File: src/app/services/cognito.service.ts (hunk 2 above)
Comment (jonwuzulike) [reply src/app/services/cognito.service.ts]: Completed.

File: src/app/routes/forgotten-password/forgotten-password.component.html (hunk 3 above)
Comment (jonwuzulike) [reply src/app/routes/forgotten-password/forgotten-password.component.html]: Completed.

**Comment by jonwuzulike** [comment]
Pushed again with fixed issues.

File: src/app/services/cognito.service.ts [hunk 4]
```diff
@@ -113,10 +115,38 @@ export class CognitoService {
...
       }
     });
-
     return resultStream;
   }
+
+/*
+  * The user will need to provide their email which Cognito for check the user
+  * pool for and then create a new password.
+  */
```
Comment (MarcinSalamon) [inline src/app/services/cognito.service.ts]: add author and params
you can use /** to generate NgDocs

File: src/app/routes/forgotten-password/forgotten-password.component.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  }
+  /**
+   * This methods allows a user to reset their password.
+   */
+  reset() {
+   const rest =  this.cognito.resetPassword(this.email);
+     this.router.navigate(['login']);
+  }
+
+
```
Comment (MarcinSalamon) [inline src/app/routes/forgotten-password/forgotten-password.component.ts]: delete extra spaces and generate NgDocs

**Review by MarcinSalamon — CHANGES_REQUESTED** [review]
generate NgDocs

**Comment by danielsidhu** [comment]
you have some conflicts with the following files that will need some refactoring. It is too complex for me to fix here with the GUI

File: src/app/services/cognito.service.ts (hunk 4 above)
Comment (jonwuzulike) [reply src/app/services/cognito.service.ts]: Completed.

File: src/app/routes/forgotten-password/forgotten-password.component.ts (hunk 5 above)
Comment (jonwuzulike) [reply src/app/routes/forgotten-password/forgotten-password.component.ts]: Completed.

**Comment by MarcinSalamon** [comment]
Pull from dev and work on the merge conflicts, I will be much faster that way

**Comment by jonwuzulike** [comment]
Created a new branch named instead.

**Pull request closed without merging**